js/src/credentialRequest/types/LdpVcCredentialRequest.js

```diff
@@ -0,0 +1,58 @@
+const {ProofType} = require("../../constants/ProofType");
+const CredentialRequest = require("../credentialRequest");
+
+class LdpVcCredentialRequest extends CredentialRequest {
+    constructor(accessToken, issuerMetaData, proof) {
+        super();
+        this.accessToken = accessToken;
+        this.issuerMetaData = issuerMetaData;
+        this.proof = proof;
+    }
+
+    constructRequest() {
+        const request = {
+          requestBody: this.generateRequestBody(),
+          requestHeader: {
+            'Authorization': `Bearer ${this.accessToken}`,
+            'Content-Type': 'application/json',
+          },
+        };
+        return request;
+      }
+
+    generateRequestBody() {
+        try {
+            const requestBody = new CredentialRequestBody(
+              this.issuerMetaData.credentialFormat,
+              new CredentialDefinition(this.issuerMetaData.credentialType),
+              {
+                proof_type: ProofType.JWT,
+                jwt: this.proof,
```

Comment:
Can we extract a class for Proof as well similar to swift / kotlin?

So if we need add more proof like cwt or so in future it can be easily extracted